Jenkinsfile

```diff
@@ -60,12 +65,37 @@ pipeline {
                 not { branch 'develop' }
                 not { branch 'master' }
             }
-            steps { 
+            steps {
                 script {
                     utils.killOldBuilds()
                 }
             }
         }
+        stage("Make-doxygen") {
+            agent any
+            when { branch 'PR-795'}
+            steps {
+                    retry(3) { checkout scm }
+                    withCredentials([usernamePassword(credentialsId: 'a630aebc-6861-4e69-b497-fd7f496ec46b',
+                        usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_PASSWORD')]) {
+                        sh """#!/bin/bash
+                            set -x
+                            make doxygen
+                            git config --global user.email "mc.stanislaw@gmail.com"
+                            git config --global user.name "Stan Jenkins"
+                            git checkout --detach
+                            git branch -D gh-pages
+                            git push https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/${fork()}/math.git :gh-pages
+                            git checkout --orphan gh-pages
+                            git rm --cached -r stan test lib make
+                            git add -f doc
+                            git commit -m "auto generated docs from Jenkins"
+                            git subtree push --prefix doc/api/html https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/${fork()}/math.git gh-pages
```

Comment:
I couldn't figure out the index redirect method so instead I did this subtree method which was inspired by these pages

https://gist.github.com/cobyism/4730490
https://gist.github.com/tduarte/eac064b4778711b116bb827f8c9bef7b

I think the subtree method is a bit nicer since it puts all the html at the top layer instead of having some a redirect

Reply:
Also for this method I had to delete the remote version of gh-pages because you cannot do a forced update with `git subtree push`

Reply:
This seems like a good solution.